superbench/analyzer/data_diagnosis.py

```diff
@@ -0,0 +1,234 @@
+                logger.error('DataDiagnosis: empty raw data')
+                return data_not_accept_df, label_df
+            # get criteria
+            if not self._get_criteria(rule_file, baseline_file):
+                return data_not_accept_df, label_df
+            # run diagnosis rules for each node
+            for node in self._raw_data_df.index:
+                details_row, summary_data_row = self._run_diagnosis_rules_for_single_node(node)
+                if details_row:
+                    data_not_accept_df.loc[node] = details_row
+                    summary_details_df = summary_details_df.append(summary_data_row)
+                    label_df.loc[node] = 1
+                else:
+                    label_df.loc[node] = 0
+            # combine details for issued nodes
+            if len(data_not_accept_df) != 0:
+                data_not_accept_df = data_not_accept_df.join(summary_details_df)
+                data_not_accept_df = data_not_accept_df.sort_values(by=summary_columns, ascending=False)
+
+        except Exception as e:
+            logger.error('DataDiagnosis: run diagnosis rules failed, message: {}'.format(str(e)))
+        return data_not_accept_df, label_df
+
+    def excel_output(self, data_not_accept_df, output_dir):
+        """Output the processed results into excel file.
+
+        Args:
+            data_not_accept_df (DataFrame): issued nodes's detailed information
+            output_dir (str): the path of output excel file
+        """
+        try:
+            writer = pd.ExcelWriter(output_dir + '/diagnosis_summary.xlsx', engine='xlsxwriter')
+            # Check whether writer is valiad
+            if not isinstance(writer, pd.ExcelWriter):
+                logger.error('DataDiagnosis: excel_data_output - invalid file path.')
+                return
+            file_handler.excel_raw_data_output(writer, self._raw_data_df, 'Raw Data')
+            file_handler.excel_data_not_accept_output(writer, data_not_accept_df, self._sb_rules)
+            writer.save()
+        except Exception as e:
+            logger.error('DataDiagnosis: excel_data_output - {}'.format(str(e)))
+
+    def run(self, raw_data_path, rule_file, output_dir, output_format='excel'):
+        """Run the data diagnosis and output the results.
+
+        Args:
+            raw_data_path (str): the path of raw data jsonl file.
+            rule_file (str): The path of baseline yaml file
+            output_dir (str): the path of output excel file
+            output_format (str): the format of the output, 'excel' or 'json'
+        """
+        try:
+            self._raw_data_df = file_handler.read_raw_data(raw_data_path)
+            self._metrics = self._get_metrics_by_benchmarks(list(self._raw_data_df.columms))
+            logger.info('DataDiagnosis: Begin to processe {} nodes'.format(len(self._raw_data_df)))
+            data_not_accept_df, label_df = self.run_diagnosis_rules(self, rule_file)
```

Comment:
Why the parameters are `self` and `rule_file`?

Reply:
Just a little curious, how to pass the param `baseline_file` from `run()` to `run_diagnosis_rules()`?

Reply:
you're right, it missed the 'baseline_file', added